lib/charms/sdcore_nms_k8s/v0/fiveg_core_gnb.py

```diff
@@ -0,0 +1,445 @@
+            return None
+
+        remote_app_relation_data = dict(relation.data[relation.app])
+
+        if not data_matches_requirer_schema(remote_app_relation_data):
+            logger.error("Invalid relation data: %s", remote_app_relation_data)
+            return None
+
+        return remote_app_relation_data
+
+    def get_gnb_name(self, relation_id: int) -> Optional[str]:
+        """Return the name of the CU/gNodeB for the given relation.
+
+        Args:
+            relation_id (int): Relation ID.
+
+        Returns:
+            str: gNodeB name.
+        """
+        if remote_relation_data := self._get_remote_app_relation_data(relation_id):
+            return remote_relation_data["gnb-name"]
+        return None
+
+
+class FivegCoreGnbRequirerAppData(BaseModel):
+    """Requirer application data for fiveg_core_gnb."""
+    gnb_name: str = Field(
+        alias="gnb-name",
+        description="CU/gNB unique identifier",
+        examples=["gnb001"],
+    )
+
+
+class RequirerSchema(DataBagSchema):
+    """Requirer schema for fiveg_core_gnb."""
+
+    app_data: FivegCoreGnbRequirerAppData
+
+
+def data_matches_requirer_schema(data: dict) -> bool:
+    """Return whether data matches requirer schema.
+
+    Args:
+        data (dict): Data to be validated.
+
+    Returns:
+        bool: True if data matches requirer schema, False otherwise.
+    """
+    try:
+        RequirerSchema(app_data=FivegCoreGnbRequirerAppData(**data))
+        return True
+    except ValidationError as e:
+        logger.error("Invalid data: %s", e)
+        return False
+
+
+class FivegCoreGnbRequires(Object):
+    """Class to be instantiated by requirer of the `fiveg_core_gnb`."""
+
+    def __init__(self, charm: CharmBase, relation_name: str):
+        """Create a new instance of the FivegCoreGnbRequires class.
+
+        Args:
+            charm: Juju charm
+            relation_name (str): Relation name
+        """
+        self.relation_name = relation_name
+        self.charm = charm
+        super().__init__(charm, relation_name)
+
+    def publish_gnb_information(self, gnb_name: str) -> None:
+        """Set CU/gNB identifier in the relation data.
+
+        Args:
+            gnb_name (str): CU/gNB unique identifier.
+        """
+        if not self.charm.unit.is_leader():
+            raise RuntimeError("Unit must be leader to set application relation data.")
+
+        if not data_matches_requirer_schema(
+            data={"gnb-name": gnb_name}
+        ):
+            raise ValueError(f"Invalid gNB name: {gnb_name}")
+
+        relation = self.model.get_relation(relation_name=self.relation_name)
+        if not relation:
+            raise RuntimeError(f"Relation {self.relation_name} not created yet.")
+
+        relation.data[self.charm.app].update({"gnb-name": gnb_name})
+
+    def _get_remote_app_relation_data(self, relation: Optional[Relation] = None) -> Optional[dict]:
```

Comment:
```suggestion
    def _get_remote_app_relation_data(self) -> Optional[dict]:
```
We're not using `relation` anywhere, so I guess we can remove it.